x-pack/legacy/plugins/ml/public/application/components/color_range_legend/use_color_range.ts

```diff
@@ -0,0 +1,188 @@
+    return t;
+  }
+
+  if (t < 1 / n) {
+    return 0;
+  }
+  if (t < 3 / n) {
+    return (n / 4) * (t - 1 / n);
+  }
+  return 0.5 + (t - 3 / n);
+};
+
+export enum COLOR_RANGE_SCALE {
+  LINEAR = 'linear',
+  INFLUENCER = 'influencer',
+  SQRT = 'sqrt',
+}
+
+/**
+ * Color range scale options in the format for EuiSelect's options prop.
+ */
+export const colorRangeScaleOptions = [
+  {
+    value: COLOR_RANGE_SCALE.LINEAR,
+    text: i18n.translate('xpack.ml.components.colorRangeLegend.linearScaleLabel', {
+      defaultMessage: 'Linear',
+    }),
+  },
+  {
+    value: COLOR_RANGE_SCALE.INFLUENCER,
+    text: i18n.translate('xpack.ml.components.colorRangeLegend.influencerScaleLabel', {
+      defaultMessage: 'Influencer custom scale',
+    }),
+  },
+  {
+    value: COLOR_RANGE_SCALE.SQRT,
+    text: i18n.translate('xpack.ml.components.colorRangeLegend.sqrtScaleLabel', {
+      defaultMessage: 'Sqrt',
+    }),
+  },
+];
+
+export enum COLOR_RANGE {
+  BLUE = 'blue',
+  RED = 'red',
+  RED_GREEN = 'red-green',
+  GREEN_RED = 'green-red',
+  YELLOW_GREEN_BLUE = 'yellow-green-blue',
+}
+
+/**
+ * Color range options in the format for EuiSelect's options prop.
+ */
+export const colorRangeOptions = [
+  {
+    value: COLOR_RANGE.BLUE,
+    text: i18n.translate('xpack.ml.components.colorRangeLegend.blueColorRangeLabel', {
+      defaultMessage: 'Blue',
+    }),
+  },
+  {
+    value: COLOR_RANGE.RED,
+    text: i18n.translate('xpack.ml.components.colorRangeLegend.redColorRangeLabel', {
+      defaultMessage: 'Red',
+    }),
+  },
+  {
+    value: COLOR_RANGE.RED_GREEN,
+    text: i18n.translate('xpack.ml.components.colorRangeLegend.redGreenColorRangeLabel', {
+      defaultMessage: 'Red - Green',
+    }),
+  },
+  {
+    value: COLOR_RANGE.GREEN_RED,
+    text: i18n.translate('xpack.ml.components.colorRangeLegend.greenRedColorRangeLabel', {
+      defaultMessage: 'Green - Red',
+    }),
+  },
+  {
+    value: COLOR_RANGE.YELLOW_GREEN_BLUE,
+    text: i18n.translate('xpack.ml.components.colorRangeLegend.yellowGreenBlueColorRangeLabel', {
+      defaultMessage: 'Yellow - Green - Blue',
+    }),
+  },
+];
+
+/**
+ * A custom Yellow-Green-Blue color range to demonstrate the support
+ * for more complex ranges with more than two colors.
+ */
+const coloursYGB = [
+  '#FFFFDD',
+  '#AAF191',
+  '#80D385',
+  '#61B385',
+  '#3E9583',
+  '#217681',
+  '#285285',
+  '#1F2D86',
+  '#000086',
+];
+const colourRangeYGB = d3.range(0, 1, 1.0 / (coloursYGB.length - 1));
+colourRangeYGB.push(1);
+
+const colorDomains = {
+  [COLOR_RANGE.BLUE]: [0, 1],
+  [COLOR_RANGE.RED]: [0, 1],
+  [COLOR_RANGE.RED_GREEN]: [0, 1],
+  [COLOR_RANGE.GREEN_RED]: [0, 1],
+  [COLOR_RANGE.YELLOW_GREEN_BLUE]: colourRangeYGB,
+};
+
+/**
+ * Custom hook to get a d3 based color range to be used for color coding in table cells.
+ *
+ * @param colorRange COLOR_RANGE enum.
+ * @param colorRangeScale COLOR_RANGE_SCALE enum.
+ * @param featureCount
+ */
+export const useColorRange = (
```

Comment:
is this a hook? the name suggests so, but it doesn't look like one inside.
If not, perhaps the name should change?

Reply:
it needs to be one because it uses another custom hook  (`useUiChromeContext()`)  on the inside.

Reply:
i didn't spot that